src/bus.js

```diff
@@ -0,0 +1,24 @@
+export default {
+    events: {},
+
+    on(name, callback) {
+        if (! this.events.hasOwnProperty(name)) {
```

Comment:
Does this work the same?

```js
if (!Array.isArray(this.events[name])) {
  this.events[name] = [ cb ];
} else {
  this.events[name].push(cb)
}
```

Reply:
Yeah that would work the same.

Reply:
Since the callback is _always_ going to be added to the array, I prefer checking the unhappy path first and setting it up, then always going through to the happy path... what do you think?

Reply:
Yeah the main issue was "create an empty array to immediately push to it" Vs "create an array with the element already in it" but totally a code style thing

Reply:
Yeah for sure, I find the current one more readable and understandable at a quick glance. I'll leave it for now :)